gateway/implementation/src/main/java/eu/openg/guessnumberapi/gateway/implementation/InMemoryGameEntityRepo.java

```diff
@@ -10,19 +11,27 @@
 
 public class InMemoryGameEntityRepo implements GameEntityRepository {
     private final Map<Integer, GameEntity> gameIdToGameEntityMap;
+    private final GameIdProvider gameIdProvider;
 
-    public InMemoryGameEntityRepo() {
-        this(new HashMap<>());
+    public InMemoryGameEntityRepo(GameIdProvider gameIdProvider) {
+        this(new HashMap<>(), gameIdProvider);
     }
 
-    public InMemoryGameEntityRepo(Map<Integer, GameEntity> storage) {
+    public InMemoryGameEntityRepo(Map<Integer, GameEntity> storage, GameIdProvider gameIdProvider) {
         gameIdToGameEntityMap = storage;
+        this.gameIdProvider = gameIdProvider;
     }
 
     @Override
-    public void save(GameEntity restGameEntity) {
-        if (nonNull(restGameEntity))
-            gameIdToGameEntityMap.put(restGameEntity.returnGameId(), restGameEntity);
+    public int save(GameEntity restGameEntity) {
+        if (nonNull(restGameEntity)) {
+            int id = restGameEntity.getGameId();
+            if (id == 0)
+                id = gameIdProvider.getNextId();
```

Comment:
tarkim id = 0 ir įeina į šitą ifą. čia gausi naują id, tarkim 1. sekančioj eilutėj į mapą įdedi naują entry, kurio key (id) = 1, tačiau id objekte = 0